rfc/command-definitions.md

```diff
@@ -0,0 +1,190 @@
+# RFC: UCAN Command Definitions
+Status: Proposed Standard
+
+## Introduction
+
+This RFC proposes a definition and formal documentation format for UCAN Commands, to replace the manually created and inconsistently structured specification documents we currently use.
+
+This is largely a proposal for UCAN itself, but as it's motivated by needs arising in Storacha, it will at least begin as a draft RFC here for initial feedback. It is fully compatible with the existing UCAN 1.0 specifications, and could be adopted by UCAN itself or live on top of it. However, its utility is limited if it is not adopted outside of a single organization.
+
+### Notation
+
+*Several elements of the proposal below will be illustrated with examples displayed in DAG-JSON. The canonical representations of such elements, however, is DAG-CBOR.*
+
+*Comments such as `// "/": "bafy..` denote the CIDs of objects.*
+
+## Problem
+
+Understanding a UCAN document requires a great deal of implicit context. UCAN provides strict semantics for determining whether an invocation of a Command was issued with authority, and should therefore be executed. It does not, however, provide strict semantics for what that execution should consist of.
+
+Consider the following scenario. Alice has some files in a home directory on a server. She'd like Bob to be able to access those files. Alice trusts Bob fully with the contents of this home directory, and so issues the following delegation:
+
+```json5
+{
+  // "/": "bafy..accessdel",
+  "iss": "did:example:alice",
+  "aud": "did:example:bob",
+  "sub": "did:example:alice",
+  "cmd": "/home/access",
+  "nonce": "…",
+  "exp": null,
+}
+```
+
+But Bob, whether out of malice or misunderstanding, then issues the following invocation:
+
+```json5
+{
+  "iss": "did:example:bob",
+  "aud": "did:example:alices-front-door",
+  "sub": "did:example:alice",
+  "cmd": "/home/access",
+  "nonce": "…",
+  "exp": null,
+  "prf": [{"/": "bafy..accessdel"}]
+}
+```
+
+`did:example:alices-front-door` is the smart lock on Alice's front door. Seeing that Bob has authority from Alice to "access" her "home", it accordingly unlocks and lets Bob in, where he has a very different kind of access than what Alice intended to delegate.
```

Comment:
Shouldn't the _subject_ differentiate here? You're not accessing "alice", you're accessing the alice's house. The house should have a DID, as should the server space - this is the subject of the invocation.